Classes/Hooks/HookService.php

```diff
@@ -0,0 +1,129 @@
+     *
+     * @var boolean
+     */
+    protected $hooksHaveBeenRetrieved = false;
+
+    /**
+     * The constructor.
+     *
+     * @throws \UnexpectedValueException
+     *         if $interfaceName does not extend Hook interface
+     *
+     * @param $interfaceName interface the hook needs implemented
+     * @param $index optional index to $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars']
+     *               if not using the interface name (for backwards compatibility)
+     *               (the interface name is recommended)
+     */
+    public function __construct(string $interfaceName, string $index = '')
+    {
+        if (!interface_exists($interfaceName)) {
+            throw new UnexpectedValueException(
+                'The interface ' . $interfaceName . ' does not exist.',
+                1565089078
+            );
+        }
+
+        if (!in_array(Hook::class, class_implements($interfaceName), true)) {
+            throw new UnexpectedValueException(
+                'The interface ' . $interfaceName . ' does not extend '
+                    . Hook::class . ' interface.',
+                1565088963
+            );
+        }
+
+        $this->interfaceName = $interfaceName;
+        $this->index = $interfaceName;
+        if ($index != '') {
+            $this->index = $index;
+        }
+    }
+
+    /**
+     * Gets the hook objects for the interface.
+     *
+     * @throws \UnexpectedValueException
+     *         if there are registered hook classes that do not implement the
+     *         $this->interfaceName interface
+     *
+     * @return array
+     *         the hook objects, will be empty if no hooks have been set
+     */
+    public function getHooks(): array
+    {
+        if (!$this->hooksHaveBeenRetrieved) {
+            $hookClasses = $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][$this->index];
+            if (is_array($hookClasses)) {
+                foreach ($hookClasses as $hookClass) {
+                    $hookInstance = GeneralUtility::makeInstance($hookClass);
+                    if (!($hookInstance instanceof $this->interfaceName)) {
+                        throw new UnexpectedValueException(
+                            'The class ' . get_class($hookInstance) . ' is registered for the ' . $this->index .
+                                ' hook list, but does not implement the ' . $this->interfaceName . ' interface.',
+                            1448901897
+                        );
+                    }
+                    $this->hookObjects[] = $hookInstance;
+                }
+            }
+
+            $this->hooksHaveBeenRetrieved = true;
+        }
+
+        return $this->hookObjects;
+    }
+}
+
+if (defined('TYPO3_MODE')
+    && isset($GLOBALS['TYPO3_CONF_VARS'][TYPO3_MODE]['XCLASS']['ext/seminars/Classes/Hooks/Hook.php'])
+) {
+    include_once($GLOBALS['TYPO3_CONF_VARS'][TYPO3_MODE]['XCLASS']['ext/seminars/Classes/Hooks/Hook.php']);
+}
```

Comment:
No legacy XCLASS footers anymore, please. They are not needed nowadays.

Reply:
Maybe I'm not informed well, but what Typo3 core mechanism replaced that?

Reply:
Autoloading replaced that.